core/src/main/java/ai/timefold/solver/core/api/score/analysis/ScoreAnalysis.java

```diff
@@ -300,6 +300,64 @@ Explanation of score (%s):
         return summary.toString();
     }
 
+    /**
+     * Provides a summary of the solution's score analysis using the {@link #summarize(int)} method
+     * in a similar way to {@link #summarize()}.
+     * It is possible to specify the maximum number of constraint matches to display per constraint by passing
+     * the desired limit as an argument.
+     *
+     * @param topLimit maximum number of constraint matches to show per constraint.
+     *        Use {@link Integer#MAX_VALUE} to show all matches.
+     */
+    @SuppressWarnings("java:S3457")
+    public @NonNull String summarize(int topLimit) {
```

Comment:
Let's not duplicate all of this code.
Surely the existing code can be extracted and generalized so that both the old method and this new method can reuse it.

The old method applies a pre-configured limit. It should not be too hard to implement that method with this method.